bubble_analyser/default.py

```diff
@@ -0,0 +1,161 @@
+import tomllib
+from pprint import pprint
+
+import matplotlib.pyplot as plt
+import numpy as np
+from scipy import ndimage
+from skimage import (
+    color,
+    filters,
+    io,
+    measure,
+    morphology,
+    segmentation,
+    transform,
+)
+from skimage.io import imsave
+
+
+def extendedmin(img, H):
```

Comment:
All and every function that you write need: 
- a docstring explaining what the function is doing. Follow the [Google standard](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) for this. 
- type annotations. Python is a dynamic language, so you do not have to indicate the type for the program to work, but more and more it is becoming part of the good practices to do so. This way the code is better documented and static type checkers, like `mypy` - which we have enabled - can validate the code. You can read more about this in `mypy` documentation. The [cheatsheet](https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html) might come handy. 

Based on the above, this function signature will look:

```python
# Import numpy types at the top
import numpy.typing as npt
...

def extendedmin(img: npt.NDArray[np.int_], H: int) -> npt.NDArray[bool]:
    """Calculate the location of the local minima of the image.

    (here you can explain a bit how that is actually done)

    Args:
        img: The image to calculate the minima for.
        H: The neighbourhood connectivity.

    Return:
        A boolean array with the same shape than the input image where
        a value of `True` indicate the pixels of a local minima.
    """
```

This needs to be done for every function you write. It is a lot of work, but if done as you are writing the code, it is way easier than having to do it afterwards.